minio/signer.py

```diff
@@ -198,7 +198,7 @@ def sign_v4(method, url, region, headers=None, access_key=None,
 
     date = datetime.utcnow()
     headers['X-Amz-Date'] = date.strftime("%Y%m%dT%H%M%SZ")
-    headers['X-Amz-Content-Sha256'] = content_sha256
+    headers['X-Amz-Content-Sha256'] = content_sha256.decode('ascii')
```

Comment:
It is but it should be handled in a way UNSIGNED_PAYLOAD should still be a string, keeping it as byte causes presigned signature requests to fail.